pkg/gui/controllers/files_controller.go

```diff
@@ -302,59 +302,128 @@ func (self *FilesController) stageAll() error {
 	return self.contexts.Files.HandleFocus()
 }
 
-func (self *FilesController) ignore(node *filetree.FileNode) error {
-	if node.GetPath() == ".gitignore" {
-		return self.c.ErrorMsg("Cannot ignore .gitignore")
-	}
+func (self *FilesController) unstageFiles(node *filetree.FileNode) error {
 
-	unstageFiles := func() error {
-		return node.ForEachFile(func(file *models.File) error {
-			if file.HasStagedChanges {
-				if err := self.git.WorkingTree.UnStageFile(file.Names(), file.Tracked); err != nil {
-					return err
-				}
+	return node.ForEachFile(func(file *models.File) error {
+		if file.HasStagedChanges {
+			if err := self.git.WorkingTree.UnStageFile(file.Names(), file.Tracked); err != nil {
+				return err
 			}
+		}
 
-			return nil
-		})
-	}
+		return nil
+	})
+}
+
+func (self *FilesController) checkTracking(node *filetree.FileNode, trText string, trPrompt string, trAction string, f func(string) error) (error, bool) {
 
 	if node.GetIsTracked() {
 		return self.c.Confirm(types.ConfirmOpts{
-			Title:  self.c.Tr.IgnoreTracked,
-			Prompt: self.c.Tr.IgnoreTrackedPrompt,
+			Title:  trText,
+			Prompt: trPrompt,
 			HandleConfirm: func() error {
-				self.c.LogAction(self.c.Tr.Actions.IgnoreFile)
+				self.c.LogAction(trAction)
 				// not 100% sure if this is necessary but I'll assume it is
-				if err := unstageFiles(); err != nil {
+				if err := self.unstageFiles(node); err != nil {
 					return err
 				}
 
 				if err := self.git.WorkingTree.RemoveTrackedFiles(node.GetPath()); err != nil {
 					return err
 				}
 
-				if err := self.git.WorkingTree.Ignore(node.GetPath()); err != nil {
+				if err := f(node.GetPath()); err != nil {
 					return err
 				}
+
 				return self.c.Refresh(types.RefreshOptions{Scope: []types.RefreshableView{types.FILES}})
 			},
-		})
+		}), true
 	}
+	return nil, false
+}
 
-	self.c.LogAction(self.c.Tr.Actions.IgnoreFile)
+func (self *FilesController) ignoreOrExcludeFile(node *filetree.FileNode, trText string, trPrompt string, trAction string, f func(string) error) error {
 
-	if err := unstageFiles(); err != nil {
+	trackingErr, earlyExit := self.checkTracking(node, trText, trPrompt, trAction, f)
+
+	if trackingErr != nil {
+		return trackingErr
+	}
+
+	if earlyExit {
+		return nil
+	}
+
+	self.c.LogAction(trAction)
+
+	if err := self.unstageFiles(node); err != nil {
 		return err
 	}
 
-	if err := self.git.WorkingTree.Ignore(node.GetPath()); err != nil {
+	if err := f(node.GetPath()); err != nil {
 		return self.c.Error(err)
 	}
 
 	return self.c.Refresh(types.RefreshOptions{Scope: []types.RefreshableView{types.FILES}})
 }
 
+func (self *FilesController) ignore(node *filetree.FileNode) error {
+	if node.GetPath() == ".gitignore" {
+		return self.c.ErrorMsg("Cannot ignore .gitignore")
```

Comment:
I know this wasn't previously internationalised but could we do that now? I.e. create an entry in english.go for this string and reference it here (same with the other errors below)

Reply:
Done :)